src/internal.c

```diff
@@ -6944,16 +6971,14 @@ static int DoCertificate(WOLFSSL* ssl, byte* input, word32* inOutIdx,
 
                     ret = ParseCertRelative(args->dCert, CERT_TYPE,
                                     !ssl->options.verifyNone, ssl->ctx->cm);
-                    if (ret != 0) {
-                    #ifdef WOLFSSL_ASYNC_CRYPT
-                        if (ret == WC_PENDING_E) {
-                            ret = wolfSSL_AsyncPush(ssl,
-                                args->dCert->sigCtx.asyncDev,
-                                WC_ASYNC_FLAG_CALL_AGAIN);
-                        }
-                    #endif
+                #ifdef WOLFSSL_ASYNC_CRYPT
+                    if (ret == WC_PENDING_E) {
+                        ret = wolfSSL_AsyncPush(ssl,
+                            args->dCert->sigCtx.asyncDev,
+                            WC_ASYNC_FLAG_CALL_AGAIN);
                         goto exit_dc;
```

Comment:
This goto exit_dc is now only in ASYNC build. That doesn't seem right.  Where is the ParseCertRelative failure handled in non async or an async failure that isn't WC_PENDING_E?